client/src/app/deck/deck-select.tsx

```diff
@@ -194,8 +208,15 @@ function SelectDeck({
 				deckExists = true
 			}
 		})
-		deckExists && setShowOverwriteModal(true)
-		!deckExists && saveDeckInternal(deck)
+		if (deckExists) {
+			setShowOverwriteModal(true)
+			return
+		}
+
+		saveDeckInternal(deck)
+		setSavedDecks(getSavedDecks())
+		setSortedDecks(sortDecks([...savedDecks, JSON.stringify(deck)]))
```

Comment:
Why are you stringifying then parsing later?

Reply:
The function takes in only strings... YEah this is really bad. I should refactor this part of the code so multiple behavior isn't handled in one function

Reply:
Please fix this... this function should not take a string.

Reply:
Resolved